crates/trigger/src/stdio.rs

```diff
@@ -56,13 +56,23 @@ impl StdioLoggingTriggerHooks {
         &self,
         component_id: &str,
         log_suffix: &str,
-        log_dir: &Path,
+        log_dir: Option<&Path>,
     ) -> Result<ComponentStdioWriter> {
         let sanitized_component_id = sanitize_filename::sanitize(component_id);
-        let log_path = log_dir.join(format!("{sanitized_component_id}_{log_suffix}.txt"));
+        let mut log_path: Option<PathBuf> = None;
+        if let Some(log_dir) = log_dir {
+            log_path = Some(log_dir.join(format!(
+                "{sanitized_component_id}_{log_suffix}.txt",
+                sanitized_component_id = sanitized_component_id,
+                log_suffix = log_suffix
+            )))
+        };
+        let log_path = log_path.as_deref();
+
         let follow = self.follow_components.should_follow(component_id);
-        ComponentStdioWriter::new(&log_path, follow)
-            .with_context(|| format!("Failed to open log file {}", quoted_path(&log_path)))
+        // It is safe to unwrap the log_path b/c new will only fail if the file can't be opened in which case log_path is Some
```

Comment:
This makes some hefty assumptions about the implementation of `ComponentStdioWriter::new`.  I'm sceptical we can ensure it remains infallible in the `None` case as other people maintain it.

A possibly safer (but wordier!) approach is to split `ComponentStdioWriter::new` into two functions e.g. `ComponentStdioWriter::log_path(Path) -> (fallible)` and `ComponentStdioWriter::stdio() -> (infallible)`.  (Maybe `new_forward` and `new_inherit` already are those functions.)